spec/classes/sogo_spec.rb

```diff
@@ -127,5 +156,34 @@
         is_expected.not_to contain_file('/etc/sysconfig/sogo')
       end
     end
+
+    context 'with multiple SOGoUserSources' do
+      let :params do
+        {
+          'config' => {
+            'SOGoSieveScriptsEnabled' => 'YES',
+            'SOGoUserSources' => [
+              {
+                'type' => 'sql',
+                'id' => 'directory',
+                'viewURL' => 'postgresql:/sogo@127.0.0.1/sogo/sogo_view',
```

Comment:
```suggestion
                'viewURL' => 'postgresql://sogo@127.0.0.1/sogo/sogo_view',
```

Reply:
Is there a reason you're testing this twice? I would prefer to move it into the generic tests section (if possible)

Reply:
The idea is to first test Debian based systems since they have to notify the service `sogo`. The second test is meant to test Redhat based systems which have to notify the service `sogod`.

I see following options to avoid a duplication:
* Do not test if the service is being notified after the update
* Overwrite the `service_name` in the params definition to use the same service name for all flavors

Let me know if you have another idea / which way you prefer.

Reply:
Oh, i didn't realize that... I think assigning a unique name to the service resource
```puppet
  service { 'sogo':
    name => $sogo::service_name,
    ...
  }
```
and moving the test to generic afterwards is the better option.

Reply:
I updated the PR according to your comment. I did a similiar change within the `envconfig` class to be able handle all unit tests completly generic without having cases for Debian / RedHat specifics.

Reply:
Thanks! :)